test/lib/recordevents/event_log.go

```diff
@@ -0,0 +1,22 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package recordevents
+
+// EventLog is the contract for an event logger to vent an event.
+type EventLog interface {
```

Comment:
why are you copying this to `test/lib`?

Reply:
now we have three copies of this code?

Reply:
Because that code is hard to reuse, I had to do some important changes (interfaces and impls looks similar, but they're different)

Reply:
That's why I'm saying, if we use that code only in recordevents, what if we just remove the one under `test/lib`?

Reply:
> now we have three copies of this code?

Copying to the new test framework was a mistake, i'm going to port this new code there :smile: